pkg/config/setup/config.go

```diff
@@ -194,104 +194,55 @@ const (
-	// If enabling log_payloads, ensure the log level is set to at least DEBUG to be able to see the logs
-	config.BindEnvAndSetDefault("log_payloads", false)
-	config.BindEnvAndSetDefault("log_file", "")
-	config.BindEnvAndSetDefault("log_file_max_size", "10Mb")
-	config.BindEnvAndSetDefault("log_file_max_rolls", 1)
-	config.BindEnvAndSetDefault("log_level", "info")
-	config.BindEnvAndSetDefault("log_to_syslog", false)
-	config.BindEnvAndSetDefault("log_to_console", true)
-	config.BindEnvAndSetDefault("log_format_rfc3339", false)
-	config.BindEnvAndSetDefault("log_all_goroutines_when_unhealthy", false)
-	config.BindEnvAndSetDefault("logging_frequency", int64(500))
-	config.BindEnvAndSetDefault("disable_file_logging", false)
-	config.BindEnvAndSetDefault("syslog_uri", "")
-	config.BindEnvAndSetDefault("syslog_rfc", false)
-	config.BindEnvAndSetDefault("syslog_pem", "")
-	config.BindEnvAndSetDefault("syslog_key", "")
-	config.BindEnvAndSetDefault("syslog_tls_verify", true)
-	config.BindEnv("ipc_address") // deprecated: use `cmd_host` instead
-	config.BindEnvAndSetDefault("cmd_host", "localhost")
-	config.BindEnvAndSetDefault("cmd_port", 5001)
-	config.BindEnvAndSetDefault("agent_ipc_host", "localhost")
-	config.BindEnvAndSetDefault("agent_ipc_port", 0)
-	config.BindEnvAndSetDefault("default_integration_http_timeout", 9)
-	config.BindEnvAndSetDefault("integration_tracing", false)
-	config.BindEnvAndSetDefault("integration_tracing_exhaustive", false)
-	config.BindEnvAndSetDefault("integration_profiling", false)
-	config.BindEnvAndSetDefault("integration_check_status_enabled", false)
-	config.BindEnvAndSetDefault("enable_metadata_collection", true)
-	config.BindEnvAndSetDefault("enable_gohai", true)
-	config.BindEnvAndSetDefault("enable_signing_metadata_collection", true)
-	config.BindEnvAndSetDefault("check_runners", int64(4))
-	config.BindEnvAndSetDefault("check_cancel_timeout", 500*time.Millisecond)
-	config.BindEnvAndSetDefault("auth_token_file_path", "")
-	config.BindEnv("bind_host")
-	config.BindEnvAndSetDefault("health_port", int64(0))
-	config.BindEnvAndSetDefault("disable_py3_validation", false)
-	config.BindEnvAndSetDefault("python_version", DefaultPython)
-	config.BindEnvAndSetDefault("allow_arbitrary_tags", false)
-	config.BindEnvAndSetDefault("use_proxy_for_cloud_metadata", false)
-	config.BindEnvAndSetDefault("remote_tagger_timeout_seconds", 30)
-
-	// Fips
-	config.BindEnvAndSetDefault("fips.enabled", false)
-	config.BindEnvAndSetDefault("fips.port_range_start", 9803)
-	config.BindEnvAndSetDefault("fips.local_address", "localhost")
-	config.BindEnvAndSetDefault("fips.https", true)
-	config.BindEnvAndSetDefault("fips.tls_verify", true)
+// initCommonWithServerless initializes configs that are common to all agents, in particular serverless.
+// Initializing the config keys takes too much time for serverless, so we try to initialize only what is reachable.
+func initCommonWithServerless(config pkgconfigmodel.Config) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏We currently don't have warnings when using an unknown config, because every single config is declared for all agents anyway.
Now that this is not the case anymore, should we warn when using unknown configs ?
It will add some overhead on a hot code path, but it would allow catching some issues earlier.